app/controllers/general_shoping_controller.rb

```diff
@@ -0,0 +1,14 @@
+class GeneralShopingController < ApplicationController
+  before_action :authenticate_user!
+  def index
+    @food_items = Food.where('quantity < 0')
+    @total_amount = 0
+    @count = 0
+    @food_items.each do |food_item|
+      food_item.quantity = food_item.quantity.negative? ? food_item.quantity.abs : food_item.quantity
+      amount = food_item.quantity * food_item.price
+      @total_amount += amount
+      @count += 1
+    end
+  end
+end
```

Comment:
- You have done well so far, but please fix the logic in your general shopping controller, it is not ideal for users to iadd a food with negative quantity. Please implement this according to the project requirements.
![eththstpin](https://github.com/jubaer919/Food-Recipe-App/assets/73607512/fb0ce31c-7f34-445f-b1c0-f8569f8f85e8)

- 
- 
-

Reply:
@mckent05 my controller already have quantity more than 0(line 4).....  fix the issues  with RecipeFood  validation....